packages/console/src/pages/ApplicationDetails/components/Settings.tsx

```diff
@@ -74,99 +79,94 @@ const Settings = ({ applicationType, oidcConfig, defaultData, isDeleted }: Props
-        className={styles.textField}
-        tooltip="application_details.post_sign_out_redirect_uri_tip"
-      >
-        <Controller
-          name="oidcClientMetadata.postLogoutRedirectUris"
-          control={control}
-          defaultValue={[]}
-          rules={{
-            validate: createValidatorForRhf(uriPatternRules),
-          }}
-          render={({ field: { onChange, value }, fieldState: { error } }) => (
-            <MultiTextInput
-              title="application_details.post_sign_out_redirect_uris"
-              value={value}
-              error={convertRhfErrorMessage(error?.message)}
-              placeholder={t('application_details.post_sign_out_redirect_uri_placeholder')}
-              onChange={onChange}
-            />
-          )}
-        />
-      </FormField>
-      <FormField
-        title="application_details.cors_allowed_origins"
-        className={styles.textField}
-        tooltip="application_details.cors_allowed_origins_tip"
-      >
-        <Controller
-          name="customClientMetadata.corsAllowedOrigins"
-          control={control}
-          defaultValue={[]}
-          rules={{
-            validate: createValidatorForRhf({
-              pattern: {
-                verify: (value) => !value || uriOriginValidator(value),
-                message: t('errors.invalid_origin_format'),
-              },
-            }),
-          }}
-          render={({ field: { onChange, value }, fieldState: { error } }) => (
-            <MultiTextInput
-              title="application_details.cors_allowed_origins"
-              value={value}
-              error={convertRhfErrorMessage(error?.message)}
-              placeholder={t('application_details.cors_allowed_origins_placeholder')}
-              onChange={onChange}
-            />
-          )}
-        />
-      </FormField>
+          tooltip="application_details.redirect_uri_tip"
+        >
+          <Controller
+            name="oidcClientMetadata.redirectUris"
+            control={control}
+            defaultValue={[]}
+            rules={{
+              validate: createValidatorForRhf({
+                ...uriPatternRules,
+                required: t('application_details.redirect_uri_required'),
+              }),
+            }}
+            render={({ field: { onChange, value }, fieldState: { error } }) => (
+              <MultiTextInput
+                title="application_details.redirect_uris"
+                value={value}
+                error={convertRhfErrorMessage(error?.message)}
+                placeholder={
+                  applicationType === ApplicationType.Native
+                    ? t('application_details.redirect_uri_placeholder_native')
+                    : t('application_details.redirect_uri_placeholder')
+                }
+                onChange={onChange}
+              />
+            )}
+          />
+        </FormField>
+      )}
+      {applicationType !== ApplicationType.MachineToMachine && (
```

Comment:
The block above is also for application type not equals to M2M, can combine these condition blocks together

Reply:
it's by design - I tried fragment but the code looks ugly